idaes/power_generation/carbon_capture/compression_system/compressor.py

```diff
@@ -542,47 +543,45 @@ def initialize(blk, state_args={}, outlvl=idaeslog.NOTSET, solver='ipopt',
                                               hold_state=True)
 
-        for t in blk.flowsheet().config.time:
-            for k, v in blk.inlet.vars.items():
-                v[t].fix()
-            for k, v in blk.outlet.vars.items():
-                v[t].unfix()
+        opt = SolverFactory(solver)
+        opt.options = optarg
 
+        for t in blk.flowsheet().config.time:
             # if there is not a good guess for efficiency or outlet pressure
             # provide something reasonable.
             eff = blk.efficiency_isentropic[t]
-            eff.fix(eff.value if value(eff) > 0.3
-                    and value(eff) < 1.0 else 0.85)
-            # for outlet pressure try outlet pressure, pressure ratio
-            # then if none of those look reasonable use a pressure ratio of 2.0
-            # to calculate outlet pressure
-            Pout = blk.outlet.pressure[t]
-            Pin = blk.inlet.pressure[t]
-            prratio = value(Pout/Pin)
-            if value(Pout / Pin) < 4.0 or value(Pout / Pin) > 1.0:
-                if value(blk.ratioP[t]) < 4.0 and value(blk.ratioP[t]) > 1.0:
-                    Pout.fix(value(Pin * blk.ratioP[t]))
-                elif prratio < 4.0 and prratio > 1.0:
-                    Pout.fix(value(prratio*Pin))
-                else:
-                    Pout.fix(value(Pin * 2.0))
-            else:
-                Pout.fix()
-
-        blk.deltaP[:] = value(Pout - Pin)
-        blk.ratioP[:] = value(Pout / Pin)
-
-        for t in blk.flowsheet().config.time:
-            blk.properties_isentropic[t].pressure.value = value(
-                blk.outlet.pressure[t]
-            )
-            blk.properties_isentropic[t].flow_mol.value = value(
-                blk.inlet.flow_mol[t])
-            blk.properties_isentropic[t].enth_mol.value = value(
-                blk.inlet.enth_mol[t] * 1.2
-            )
-            blk.outlet.flow_mol[t].value = value(blk.inlet.flow_mol[t])
-            blk.outlet.enth_mol[t].value = value(blk.inlet.enth_mol[t] * 1.2)
+            eff.fix(eff.value if value(eff) > 0.3 and value(eff
+                                                            ) < 1.0 else 0.85)
+            # check for alternate pressure specs
+            if blk.outlet.pressure[t].fixed:
+                blk.ratioP[t] = value(
+                    blk.outlet.pressure[t] / blk.inlet.pressure[t])
+            elif blk.control_volume.deltaP[t].fixed:
+                blk.ratioP[t] = value(
+                    (blk.control_volume.deltaP[t] + blk.inlet.pressure[t]
+                     ) / blk.inlet.pressure[t])
+            elif blk.ratioP[t].fixed:
+                blk.outlet.pressure[t] = value(
+                    blk.ratioP[t] * blk.inlet.pressure[t])
+
+            # initialize the isentropic state block
+            isen_state_args = blk.properties_isentropic[t].define_state_vars()
+            blk.properties_isentropic.initialize(isen_state_args)
```

Comment:
This is not doing what you want it to do - in fact I am not sure what it is doing.

What you want to do here is something similar to:

```
isen_state = {}
t_init = blk.flowsheet().config.time.first()
for i in blk.properties_isentropic[t_init].define_state_vars().keys():
    state_var = getattr(blk.control_volume.properties_out[t_init], i)
    if state_var.is_indexed:
        isen_state[i] = {}
        for j in state_var:
            isen_state[i][j] = state_var[j].value
    else:
        isen_state[i] = state_var.value

blk.properties_isentropic.initialize(state_args=isen_state, solver=solver, optarg=optarg)
```

Reply:
Also note, you might want to do something similar for the outlet state. Currently it is initialized at the same conditions as the inlet, but seeing as we at least have a good idea about pressure, you should probably initialize at the expected outlet pressure instead.

To do this, rather than using the `control_volume.initialize`, you should instead call the initialize methods separately on `properties_in` and `properties_out`, providing each with good initial guesses.

Reply:
Hi @andrewlee94 : I got the below error at the outlet enthalpy calculation when I use the initial code for the isentropic state. Can you let me know how to fix it? If I keep the previous syntax, this error does not show up. Thanks!

ERROR    pyomo.core:compressor_v8.py:602 evaluating object as numeric value: fs.unit.control_volume.properties_in[0.0].enth_mol + (fs.unit.properties_isentropic[0.0].enth_mol - fs.unit.control_volume.properties_in[0.0].enth_mol)/fs.unit.efficiency_isentropic[0.0]
    (object: <class 'pyomo.core.expr.numeric_expr.SumExpression'>)
unsupported operand type(s) for +: 'int' and 'dict'

Reply:
That would seem to indicate a deeper issue, but first try doing `value(state_var)` instead of `state_var.value` in both places (indexed and unindexed states).

However, this is seemingly saying that one of the state variables (enthalpy) is an `Expression` and not a `Var`, which would seem to indicate a problem.

Reply:
I'm not sure, but it says you are trying to add and int to a dictionary.  I can't really tell where this is coming from though.  Not sure if if has anything to do with something expected to be a variable being a dict.  You may have set the value of a variable to a dictionary.

Reply:
Hi @andrewlee94 and @eslickj : I try those suggestions but they do not work. I think that is coming from the isentropic enthalpy calculation because it is not a variable (at least I did not see that variable in Pressure Changer model). Do you have any idea to fix this problem? Thanks.

Reply:
Well, the simple answer is that you cannot fix the value of an `Expression`, which is what the entropy is - you would have calculate a value for the entropy and then use that to estimate the associated state variables.

However, this is somewhat property package specific, as different property packages use different state variables. The simplest approach, and the one I think we use elsewhere, is to assume isothermal behavior for initialization (or something equivalent).

Reply:
I think a good approach to initialization here is to just use the pressure changer initialization once you have the inlet, the efficiency and pressure ration reasonable.  Just deactivate all extra constraints added in this model fix the efficiency, call the pressure changer initialization.  Then either reactivate all the constraints and free the efficiency and solve if possible, or reactivate them in a step-wise way.